### ISSUE TITLE
timezone not taken into account

### ISSUE THREAD

**Issue by PackElend** [ligi/PassAndroid#178]
I noticed that time till event is not localized although time on ticket is shown correctly

**Comment by simontb** [comment]
For me it works correctly. Time of event is 5pm and now at 10am local time it says in 7 hours. However at 10:01am it says 6 hours, so minutes are just left out.

**Comment by PackElend** [comment]
doesn't work at my Fairphone 2 running LineageOS 14.1, see below
![image](https://user-images.githubusercontent.com/3542031/37895411-825f465e-30e1-11e8-9e8b-9677fbd4c0a2.png)

**Comment by ligi** [comment]
I think the difference is not in the source of the app but the pass in question. Often when analyzing passes for problems like this the time was written in plain text in the app - then PassAndroid cannot know what TimeZone was used. To make this feature work - the time needs to be e.g. the relevantTime field of the pass and be formatted as a TimeStamp. If you can send me a pass that is not working then I can have a deeper look. The json from inside is also enough - there you can also redact the information you do not want to disclose. I only really need the part with the time in question.

**Comment by PackElend** [comment]
[share.zip](https://github.com/ligi/PassAndroid/files/1847179/share.zip)
[53557bdb-6e87-43f6-b50b-b4fd7ff1be2f.zip](https://github.com/ligi/PassAndroid/files/1847192/53557bdb-6e87-43f6-b50b-b4fd7ff1be2f.zip)
can you one of them?

**Comment by ligi** [comment]
thanks - exactly the mentioned problem the time is just plain text - no context:
`
[{"key":"hdate","label":"29.09.17, 17:30","value":"","textAlignment":"PKTextAlignmentNatural","relative":false}]`

if you are keen to fix this - please contact the supplier of this pass - I have no idea how I could fix this on the edge - ideas welcome - but I think this is not really possible ..

**Comment by simontb** [comment]
I opened share.zip and `relevantDate` property is '2017-09-29T17:15+00:00'. So if you're in UTC+2, you will see 19:15.

**Comment by ligi** [comment]
I think share.zip is not the original pass but exported from PassAndroid - I guess 53557bdb-6e87-43f6-b50b-b4fd7ff1be2f.zip is the original and the root of the problem. @PackElend - correct me when I am wrong.

**Comment by ligi** [comment]
ah - @simontb yea now I see the relevantDate is also in the original - so the pass does not really look good 
`"relevantDate":"2017-09-29T17:15+00:00"`

and
`
{"key":"hdate","label":"29.09.17, 17:30","value":"","textAlignment":"PKTextAlignmentNatural","relative":false}`

So I think the bug is in relevantDate - don't think 00:00 is correct ..-)

**Comment by PackElend** [comment]
share.zip is exported from PassAndroid
53557bdb-6e87-43f6-b50b-b4fd7ff1be2f.zip is the original

**Comment by simontb** [comment]
But the content of pass.json is identical in share.zip. Since Zürich is in MEZ, it makes sense (and in my opinion is correct behaviour by PassAndroid) that 2 hours are added.

**Comment by PackElend** [comment]
Hallo,
Have you made any progress?
As far as I understand the pass isn't of good quality is it?

**Comment by simontb** [comment]
@bmrussell could you also share your pass here? If it contains sensitive information, the JSON is also enough and you can delete sensitive parts.

**Comment by bmrussell** [comment]
Yeah sure, how do I export the JSON? I couldn't see an option (apologies if I'm being thick)

**Comment by simontb** [comment]
Just change the file extension to .zip and extract it.

**Comment by bmrussell** [comment]
Cool, here you go.
[pass.zip](https://github.com/ligi/PassAndroid/files/1970544/pass.zip)

**Comment by PackElend** [comment]
Hallo,
I'm afraid that there will be more apps coming up with the same issue.
Why not allowing a per pass offset?

**Comment by PackElend** [comment]
Didn't know that trick :)
How about lowering necessary action to change by allowing to change setting
when tabbing long on an entry?

**Comment by simontb** [comment]
I would refrain from doing so. If I got you correctly, you want an option like "don't calculate time according to current timezone". Because this is your problem. Although the issue title says "timezone not taken into account", the opposite is the case here. But in my opinion this would add unnecessary complexity to PassAndroid, because it would need to store how to handle time for each pass, and also not guarantee showing the correct time, as the pass might store it in many different ways that lead to a wrong displaying of the time.

So I suggest not fixing this on PassAndroid, but you can manually correct time for the passes that do not have the right offset, or get in touch with the pass provider to fix the issue there. Are many passes affected? So far I didn't get a single one that had a wrong time offset.
